src/display/editor/tools.js

```diff
@@ -716,6 +716,13 @@ class AnnotationEditorUIManager {
       realScale: PixelsPerInch.PDF_TO_CSS_UNITS,
       rotation: 0,
     };
+
+    if (typeof PDFJSDev === "undefined" || PDFJSDev.test("TESTING")) {
+      this._pasteEditorCallback = ({ details: { type, data } }) => {
+        this.currentLayer.pasteEditor(type, data);
+      };
+      this._eventBus._on("pasteeditor", this._pasteEditorCallback);
+    }
```

Comment:
Do we really *need* these special code-paths, since that'll never actually happen in "real" editing?
Could we not utilize the clip-board directly instead, and have the test do "the same thing" that a user may actually do?

When using the viewer the following work-flow seems to behave as you'd expect:
 - Copy an image available locally.
 - Enable Stamp-editing.
 - Press <kbd>Ctrl</kbd>+<kbd>V</kbd>

Hence, could we not *simulate* that in the integration-test instead:
 - Load the image, the same way as you already do below.
 - Add the image-data to the clip-board.
 - Enable Stamp-editing.
 - Trigger pasting, the same way that other integration-tests already do.

Reply:
It wasn't working with Firefox, I got this error: `ClipboardItem is not defined`.
Finally I found that it works with `dom.events.asyncClipboard.clipboardItem` set to `true`.